packages/studio-be/src/common/cloud-client.ts

```diff
@@ -27,12 +29,26 @@ class CloudClientError extends VError {
 }
 
 export class CloudClient {
-  private cdmAxios: AxiosInstance
-  private oauthAxios: AxiosInstance
+  private axios: AxiosInstance
 
-  constructor(private logger: Logger) {
-    this.cdmAxios = axios.create({ baseURL: `${process.CLOUD_CONTROLLER_ENDPOINT}/v1` })
-    this.oauthAxios = axios.create({ baseURL: process.CLOUD_OAUTH_ENDPOINT })
+  constructor(props: { cloudControllerEndpoint: string }) {
+    const { cloudControllerEndpoint } = props
+    this.axios = axios.create({ baseURL: `${cloudControllerEndpoint}/v1` })
+  }
+
+  public async listWorkspaces(props: {
+    personalAccessToken: PersonalAccessToken
+    signal: AbortSignal
+  }): Promise<CDMWorkspace[]> {
+    const { personalAccessToken, signal } = props
+
+    const config = this.getCloudAxiosConfig({ token: personalAccessToken, principals: {} })
+    const { data: workspaces } = await this.axios.get('/workspaces', {
```

Comment:
pretty sure you can type an axios call instead of a hard cast when returning `axios.get<T>`

Reply:
Fixed in d1bec6fa8d826eae7d659ccf985b47d89117a869